cypress/e2e/settings.spec.ts

```diff
@@ -245,6 +245,130 @@ describe("Settings page", () => {
     cy.get("@showReach").should("not.be.checked")
   })
 
+  it("Should toggle Privatise staging site button and automatically generate a password", () => {
+    // Arrange
+    // NOTE: Initial state is privatise staging off
+    cy.contains("Privatise staging site")
+      .parent()
+      .parent()
+      .find("input")
+      .as("privatiseStaging")
+    cy.get("@privatiseStaging").should("not.be.checked")
+
+    // Act
+    // Trigger click event
+    // NOTE: We have to force as chakra uses an invisible input as a checkbox
+    cy.get("@privatiseStaging")
+      .check({ force: true, timeout: 0 })
+      .then(() => {
+        // Checking within then because of wonky focus preventing the field from being populated
+        cy.contains("You should change this every 90 days")
+          .parent()
+          .next()
+          .find("input")
+          .as("passwordInput")
+        cy.wait(1000)
```

Comment:
what is this wait for ?

Reply:
I was getting wonky results when attempting to inspect the contents of the password field immediately for some reason - I think it might have been because of the auto generation + population of the field taking some time. I've opted to add in a wait time here as this made the test less flaky, but open to suggestions on making this better!

Reply:
got it

Reply:
i can work with you on this - this shouldn't block the PR cos it works fine as-is but it is an issue because static wait times, if propagated, either waits for no reason (when wait time is insufficient) or causes extra delay (resolve early prior to wait). 

the trick to solve this is to know _what_ signals completion and either assert on that or `wait` on the request